### ISSUE TITLE
Plugin key auth error

### ISSUE THREAD

**Issue by odelvalle** [Kong/kong#5043]
### Summary

Plugin key-auth throw error if key value is empty in header. This should response 401, because key value is invalid.

### Steps To Reproduce

1. Create any service
2. Add key-auth plugin to service
3. Key name: "key"
4. Make a request with header "key" without value

### Additional Details & Logs

- Kong version (`1.0.2`)
- Kong error logs 
```
Error log: [error] 39#0: *8695285 [kong] handler.lua:159 [key-auth] failed to get from node cache: [postgres] length must be at least 1
```

**Comment by rpartha** [comment]
I'll take this up if you don't mind?

**Comment by jeremyjpj0916** [comment]
This should be fixed on Kong 1.4rc2:

https://github.com/Kong/kong/blob/master/kong/plugins/key-auth/handler.lua#L140

checks for empty header:

https://github.com/Kong/kong/commit/600b12bb4b0357764e74700d8dd45a9939198391#diff-6baf8a52cf8f16b161088b8fad9cb8dc

Dup of original issue: https://github.com/Kong/kong/issues/4405
Original PR: https://github.com/Kong/kong/pull/4418

**Comment by bungle** [comment]
Thanks @jeremyjpj0916, closing this now. @odelvalle, please reopen if you still have issues with it.
